### PR TITLE
Rework models code to use the backend, and other improvements to it

### LINKED ISSUE #82: Update Models code to use backend
Related to #81

### PR DESCRIPTION
**Pull request by mhauru**
This removes all the bits from both the Arima and HODMD model codes that directly read from/wrote to the database, and replaces them with calls to the backend API. While I was at it and learning my way around the code I also made other small changes.

The diff can be made nicer once #175 has been merged.

Closes #82

### CODE REVIEW

**Comment by mhauru** [comment]
Moved the `login` and `auth_backend_call` functions to `core`, as discussed on a call.

File: dtbase/models/arima/run_pipeline.py [hunk 1]
```diff
@@ -121,41 +94,37 @@ def run_pipeline(session: Optional[Session] = None) -> None:
...
-
-
-def main() -> None:
-    run_pipeline()
+            response = auth_backend_call(
+                "post",
+                "/model/insert-model-run",
+                {
+                    "model_name": "Arima",
+                    "scenario_description": "Business as usual",
```
Comment (EdwinB12) [inline dtbase/models/arima/run_pipeline.py:114]: Should be  scenario name variable.

File: dtbase/models/hodmd/run_pipeline.py [hunk 2]
```diff
@@ -45,79 +34,89 @@ def fetch_data() -> Tuple[Dict, pd.DataFrame]:
...
+        "post",
+        "/model/insert-model-scenario",
+        {"model_name": "HODMD", "description": scenario},
+        token=token,
+    )
+    if response.status_code not in {201, 409}:
+        raise BackendCallError(response)
+
+    # Ensure that we have all measures in the database, or insert if not.
+    # We should have mean, upper bound, and lower bound for each of the measures that
```
Comment (EdwinB12) [inline dtbase/models/hodmd/run_pipeline.py:72]: Remove comment as its likely copied from arima

File: dtbase/tests/conftest.py [hunk 3]
```diff
@@ -279,6 +279,24 @@ def auth_frontend_client(conn_frontend_client: FlaskClient) -> FlaskClient:
     return conn_frontend_client
 
 
+@pytest.fixture()
+def conn_models_backend(client: AuthenticatedClient) -> Generator[None, None, None]:
```
Comment (EdwinB12) [inline dtbase/tests/conftest.py:283]: Change name to conn_backend

**Review by EdwinB12 — APPROVED** [review]

**Pull request merged**